### DISCUSSION TITLE
reconciling attachments that are not included in exports

### DISCUSSION THREAD

**Discussion by unphased** [ReagentX/imessage-exporter#191] [Q&A] [answered]
I see that after exporting everything I have under 8000 files in attachments in the export, but 30k files in my Library/Messages attachments folder. So I know that live photos are getting their little mov files thrown out, which I'm cool with (I think) but this doesn't quite seem to account for it. 

Now before I go nuclear and generate blake3 hashes on all files to get to the bottom of it, I did at least see that just listing *file* files brings the count down to 
  
    $ find /Users/slu/Library/Messages/Attachments/ -type f | wc -l      
      13902

Not really sure why Finder is saying 30k files but I guess the .pluginPayloadAttachment files in here may account for the discrepancy.

    $ find /Users/slu/Library/Messages/Attachments/ -type f | grep -v \.pluginPayloadAttachment$ | wc -l            
      11494

However it's a bit harder to find out more for sure, since it isnt easy to discriminate real movie attachments from live photo clips to filter those out... and all the attachments also got regenerated guid names so i would need to hash them to positively map them...

**Comment by ReagentX** [comment, marked as answer]
This software looks at the attachments table in the database, not at the files on the disk, which explains the discrepancy.

**Reply by unphased** [reply]
Yeah that makes sense of course. I do wonder if you could comment on why some attachments may exist that were not referenced from the database. Other than left over from other user accounts that were logged in, which wouldn't be the case here.
